util/src/main/java/org/killbill/billing/util/dao/HistorySqlDao.java

```diff
@@ -39,8 +41,10 @@
     public List<EntityHistoryModelDao<M, E>> getHistoryForTargetRecordId(@Define("bypassMappingRegistryCache") final boolean bypassMappingRegistryCache,
                                                                          @Bind("targetRecordId") final long targetRecordId,
                                                                          @SmartBindBean InternalTenantContext context);
-    @SqlUpdate
-    @GetGeneratedKeys
-    public Long addHistoryFromTransaction(@EntityHistoryBinder EntityHistoryModelDao<M, E> history,
-                                          @SmartBindBean InternalCallContext context);
+
+    @SqlBatch
+    @BatchChunkSize(1000) // Arbitrary value, just a safety mechanism in case of very large datasets
```

Comment:
That seems like a large value - i think we will need to test the perf associated with these batches. I have the same concern with my notification work. The main concern i have is that the larger the value i suspect the longer the operation, and i am worried about concurrent operations waiting on the table lock(s). It's more a con cern for the notification work though.